internal/eval/eval.go

```diff
@@ -2672,3 +2672,75 @@ func evalPFMERGE(args []string, store *dstore.Store) []byte {
 
 	return clientio.RespOK
 }
+
+func evalJSONSTRLEN(args []string, store *dstore.Store) []byte {
+	if len(args) < 1 {
+		return diceerrors.NewErrArity("JSON.STRLEN")
+	}
+
+	key := args[0]
+
+	if len(args) < 2 {
+		// no recursive
+		// making consistent with arrlen
+		// to-do parsing
+		obj := store.Get(key)
+
+		if obj == nil {
+			return clientio.RespNIL
+		}
+		jsonData := obj.Value
+
+		if utils.GetJSONFieldType(jsonData) != utils.StringType {
+			return diceerrors.NewErrWithFormattedMessage(diceerrors.JSONPathValueTypeErr)
+		}
+		return clientio.Encode(len(jsonData.(string)), false)
+	}
+
+	path := args[1]
+
+	obj := store.Get(key)
+
+	if obj == nil {
+		return clientio.RespNIL
+	}
+
+	// Check if the object is of JSON type
+	errWithMessage := object.AssertTypeAndEncoding(obj.TypeEncoding, object.ObjTypeJSON, object.ObjEncodingJSON)
+	if errWithMessage != nil {
+		return errWithMessage
+	}
+
+	jsonData := obj.Value
+	if path == defaultRootPath {
+		defaultStringResult := make([]interface{}, 0, 1)
+		if utils.GetJSONFieldType(jsonData) == utils.StringType {
+			defaultStringResult = append(defaultStringResult, int64(len(jsonData.(string))))
+		} else {
+			defaultStringResult = append(defaultStringResult, clientio.RespNIL)
+		}
+
+		return clientio.Encode(defaultStringResult, false)
+	}
```

Comment:
a tiny question. in this logic. **if path == defaultRootPath** . should the jsonData  not be a string?  because we have made judge in **AssertTypeAndEncoding**

Reply:
In AssertTypeAndEncoding, we get no error if the root value is json, which can be {a:1} or '"hello"'. If its the first case, then the defaultpath must return nil, but if its the latter it should return the length. Hence this is the reason I am doing a check here

Reply:
In AssertTypeAndEncoding, we get no error if the root value is json, which can be {a:1} or '"hello"'. If its the first case, then the defaultpath must return nil, but if its the latter it should return the length. Hence this is the reason I am doing a check here

Reply:
in which scenario. the rootpath value could be {a:1} or '"hello"
`

      127.0.0.1:6379> JSON.SET user3 $ '{a:1}'
      (error) Existing key has wrong Redis type
      127.0.0.1:6379> SET user3 '{a:1}'
      OK
      127.0.0.1:6379> JSON.STRLEN user3
      (error) Existing key has wrong Redis type
      127.0.0.1:6379> JSON.STRLEN user3 $
      (error) Existing key has wrong Redis type
      127.0.0.1:6379> JSON.SET user3 $ "hello"
      (error) Existing key has wrong Redis type
      127.0.0.1:6379> set user3 "hello"
      OK
      127.0.0.1:6379> JSON.STRLEN user3
      (error) Existing key has wrong Redis type
      127.0.0.1:6379> JSON.STRLEN user3 $
      (error) Existing key has wrong Redis type
    


`

Reply:
Hi, please check these scenarios. My bad, I did not format the cases correctly.
![image](https://github.com/user-attachments/assets/81ec27a6-4002-4c76-84dd-462c6883e8fd)
![image](https://github.com/user-attachments/assets/26ccee00-275f-4dcf-af09-9b0f8ccd767a)

Reply:
got it. thanks